probe_src/probe_frontend/README.md

```diff
@@ -0,0 +1,121 @@
+
+# PROBE Frontend
```

Comment:
That's your call, I was trying to document my code (because who else is gonna do it), but if you think it belongs at the project top-level I'm 100% fine with that. But if you don't mind, I'd like to do that in a follow-up PR (I'm already planning one to improve and unify the nix code at the top-level) since this PR very deliberately doesn't touch anything outside `probe_src/probe_frontend`.